yajsapi/executor/index.ts

```diff
@@ -465,17 +470,16 @@ export class Executor implements ComputationHistory {
             continue;
           }
           if (score < SCORE_NEUTRAL) {
+            const reason = "Score too low";
             try {
-              const reason = "Score too low";
               await proposal.reject(reason);
-              emit(new events.ProposalRejected({
-                prop_id: proposal.id(),
-                reason: reason,
-              }));
             } catch (error) {
-              //suppress and log the error and continue;
               logger.log("debug", `Reject error: ${error}`);
             }
+            emit(new events.ProposalRejected({
+              prop_id: proposal.id(),
+              reason: reason,
+            }));
```

Comment:
I'd prefer if `ProposalRejected` were emitted only when `proposal.reject()` succeeds. If it fails then we can emit `ProposalFailed`. This would be consistent with emitting `ProposalResponded` when `proposal.respond()` succeeds and `ProposalFailed` otherwise.

Reply:
Fixed